processor.go

```diff
@@ -33,6 +33,13 @@ type processor struct {
 	shuttingDown uint32
 
 	logger.Logger
+
+	tenantRegistry sync.Map
```

Comment:
Had no idea about the `sync.Map` type. Is this safe to use?

Reply:
it's like a map with RWMutex built-in. I'm not 100% sure we need this here, but since this is in a request handler, basically - probably. Doesn't hurt in any case.